setup.py

```diff
@@ -73,6 +73,8 @@
         'arangodb': ['python-arango'],
         'azurite': ['azure-storage-blob'],
         'opensearch': ['opensearch-py'],
+        'cassandra': ['cassandra-driver'],
+        'scylla': ['scylla-driver'],
```

Comment:
It looks like this dependency is not used?

Reply:
They differ in implementation, logging of both products isn't identical, also configuration isn't identical.

As for the drivers we need them both, depends on what one wants to test with

Reply:
It looks like the `ScyllaContainer` directly uses the Cassandra client here.

https://github.com/testcontainers/testcontainers-python/pull/167/files#diff-01edf1d9738ed56417c0881509e1db05ac47b51df3015ed2cee7bc5ed54a2c8dR44

Reply:
Both drivers use the same namespace, since scylla one is a drop-in replacement for the cassandra one.

I still think users should be able to choose between them in a documented way.

Reply:
Unfortunately, while `cassandra-driver` and `scylla-driver` have different pypi names, they have the same package name. They should thus not be installed together because the behavior is non-deterministic: Which package is used depends on the installation order. Consider the following scripts.

```python
# show.py: Shows the __init__.py location and md5 digest of the file.
import cassandra
import hashlib

with open(cassandra.__file__, "rb") as fp:
    digest = hashlib.md5(fp.read()).hexdigest()

print(cassandra.__file__, digest)
```

```bash
# Install the packages in different orders and compare the hashes.
echo Cassandra only.
pip uninstall -y scylla-driver cassandra-driver > /dev/null 2> /dev/null
pip install cassandra-driver > /dev/null
python playground/show.py

echo Scylla only.
pip uninstall -y scylla-driver cassandra-driver > /dev/null 2> /dev/null
pip install scylla-driver > /dev/null
python playground/show.py

echo Cassandra and Scylla.
pip uninstall -y scylla-driver cassandra-driver > /dev/null 2> /dev/null
pip install cassandra-driver > /dev/null
pip install scylla-driver > /dev/null
python playground/show.py

echo Scylla and Cassandra.
pip uninstall -y scylla-driver cassandra-driver > /dev/null 2> /dev/null
pip install scylla-driver > /dev/null
pip install cassandra-driver > /dev/null
python playground/show.py
```

```
# output
Cassandra only.
/usr/local/python/3.10.4/lib/python3.10/site-packages/cassandra/__init__.py ff2ebcf0b1df99a66633b96df60eccd8
Scylla only.
/usr/local/python/3.10.4/lib/python3.10/site-packages/cassandra/__init__.py cf43e7f36e8479d6403aec8d128e0990
Cassandra and Scylla.
/usr/local/python/3.10.4/lib/python3.10/site-packages/cassandra/__init__.py cf43e7f36e8479d6403aec8d128e0990
Scylla and Cassandra.
/usr/local/python/3.10.4/lib/python3.10/site-packages/cassandra/__init__.py ff2ebcf0b1df99a66633b96df60eccd8
```

I appreciate end users may want to install different packages depending on their use case, but we can unfortunately not install both in our CI because only one of them will be used. It looks like Scylla is a derivative of Cassandra. I'd suggest dropping the `scylla-driver` and `ScyllaContainer` and adding a note to the docstring that scylla can be used by modifying the image name (and optionally installing a different driver).

Reply:
Their docker image still behaves different, config in different ways, have different command line arguments

And different logs, current implementation uses logs to figure it's ready.

For CI we can install only one of the drivers.

If I need to implement my own code to use Scylla, then this PR doesn't have value for me, the whole purpose was to ease the testing story of both cassandra and scylla users.

Reply:
I appreciate that the two containers are different, but we don't need two separate classes because the functionality is the same in this library. E.g., the same log message is being checked [here](https://github.com/testcontainers/testcontainers-python/pull/167/files#diff-1f57e9d177cbc6ae423f97462c5869afb98e6343e2e21aea3f9674a6ba09dd4fR32) and [here](https://github.com/testcontainers/testcontainers-python/pull/167/files#diff-01edf1d9738ed56417c0881509e1db05ac47b51df3015ed2cee7bc5ed54a2c8dR33).

Reply:
If one class would inherit for the other, it would o.k. ?

It does differ in command line argument

Reply:
Yes, the command line arguments may differ, but the implementation does not make use of different command line arguments. There is no functional difference between the two implementations that I can see other than the name of the image (see [here](https://www.diffchecker.com/vrdoCzA5/)).

Reply:
seems like we are almost there, can we give a last push and finish with it, @fruch? seems like very useful to me.

Reply:
are we going to do something with it or abandoning this?